src/core/toga/widgets/switch.py

```diff
@@ -9,85 +11,189 @@ class Switch(Widget):
         and False (off, unchecked).
 
     Args:
-        label (str): Text to be shown next to the switch.
+        text (str): Text to be shown next to the switch.
         id (str): AN identifier for this widget.
         style (:obj:`Style`): An optional style object.
             If no style is provided then a new one will be created for the widget.
-        on_toggle (``callable``): Function to execute when pressed.
-        is_on (bool): Current on or off state of the switch.
+        on_change (``callable``): Function to execute when pressed.
+        value (bool): Current on or off state of the switch.
         enabled (bool): Whether or not interaction with the button is possible, defaults to `True`.
         factory (:obj:`module`): A python module that is capable to return a
             implementation of this class with the same name. (optional & normally not needed)
     """
 
     def __init__(
             self,
-            label,
+            text=None,  # BACKWARDS COMPATIBILITY: change to positional argument when the deprecated `label` is removed
             id=None,
             style=None,
-            on_toggle=None,
-            is_on=False,
+            on_change=None,
+            value=False,
             enabled=True,
             factory=None,
+            label=None,  # DEPRECATED!
+            on_toggle=None,  # DEPRECATED!
+            is_on=None,  # DEPRECATED!
     ):
         super().__init__(id=id, style=style, factory=factory)
 
         self._impl = self.factory.Switch(interface=self)
 
-        self.label = label
-        self.on_toggle = on_toggle
-        self.is_on = is_on
+        # 2022-07: Backwards compatibility
+        # label replaced with text
+        if label is not None:
+            if text is not None:
+                raise ValueError(
+                    "Cannot specify both `label` and `text`; `label` has been deprecated, use `text`"
+                )
+            else:
+                warnings.warn(
+                    "Switch.label has been renamed Switch.text", DeprecationWarning
+                )
+                text = label
+        # on_toggle replaced with on_change
+        if on_toggle:
+            if on_change:
+                raise ValueError(
+                    "Cannot specify both `on_toggle` and `on_change`; `on_toggle` has been deprecated, use `on_change`"
+                )
+            else:
+                warnings.warn(
+                    "Switch.on_toggle has been renamed Switch.on_change", DeprecationWarning
+                )
+                on_change = on_toggle
+        # is_on replaced with value
+        if is_on is not None:
+            # Note: since `value` has a default, there is no defense against setting both `value` and `is_on`
```

Comment:
We can add that defence though; you use a placeholder other than None, and check for that.